packages/explorerkit-server/src/components/idls.ts

```diff
@@ -0,0 +1,63 @@
+import { SolanaFMParser } from "@solanafm/explorer-kit";
+import { getProgramIdl, IdlItem } from "@solanafm/explorer-kit-idls";
+
+import { getSharedDep } from "@/core/shared-dependencies";
+
+export type IdlsMap = Map<string, SolanaFMParser | null>;
+
+const IDL_CACHE_TTL = 3600; // one hour
+
+export async function loadAllIdls(programIds: string[]): Promise<IdlsMap> {
+  const idls: IdlsMap = new Map();
+  const cache = getSharedDep("cache");
+  const cachedIdls = await Promise.allSettled(programIds.map((id) => cache.get(id)));
+
+  await Promise.allSettled(
+    cachedIdls
+      .map((res, i) => {
+        const programId = programIds[i]!;
+
+        if (res.status === "fulfilled" && res.value) {
+          const idl = deserializeIdl(res.value);
+
+          idls.set(programId, idl && new SolanaFMParser(idl, programId));
+
+          return;
+        }
+
+        return programId;
+      })
+      .map(async (programId) => {
+        if (!programId) {
+          return;
+        }
+
+        const idl = await getProgramIdl(programId);
+        void cache.set(programId, serializeIdl(idl), { EX: IDL_CACHE_TTL });
+        idls.set(programId, idl && new SolanaFMParser(idl, programId));
+      })
```

Comment:
I know this is existing code, but is there a reason to do this in 2 maps? Wouldn't this be more readable as `if (res.status === "fulfilled" && res.value) { ... } else { /* fetchProgra,Idl */ }`